connectors/sources/s3.py

```diff
@@ -57,17 +58,27 @@ def __init__(self, configuration):
             retries={"max_attempts": self.configuration["max_attempts"]},
         )
         self.enable_content_extraction = self.configuration["enable_content_extraction"]
+        self.s3_region_client = {}
+
+    async def client(self, region=None):
+        """This method creates client object.
+        Args:
+            region (str): Name of bucket region. Defaults to None
+        """
+        region_name = region if region else "default"
 
-    @asynccontextmanager
-    async def client(self, **kwargs):
-        """This method creates client object."""
         if AWS_ENDPOINT is not None:
             logger.debug(f"Creating a session against {AWS_ENDPOINT}")
 
-        async with self.session.client(
-            service_name="s3", config=self.config, endpoint_url=AWS_ENDPOINT, **kwargs
-        ) as s3:
-            yield s3
+        if not (region_name in self.s3_region_client):
+            s3_client = await self.session.client(
+                service_name="s3",
+                config=self.config,
+                endpoint_url=AWS_ENDPOINT,
+                region_name=region,
+            ).__aenter__()
```

Comment:
I have added a close method in this class which calls `close()` method and it  internally calls __aexit__() method of the same class , [https://github.com/aio-libs/aiobotocore/blob/master/aiobotocore/httpsession.py#L170`](https://github.com/aio-libs/aiobotocore/blob/master/aiobotocore/httpsession.py#L170%60)

Reply:
I'm not sure about this change - instead of manually calling `__aenter__` and `__exit__` indirectly, why not making it a context method with such signature:

```
def with_client(self, region=None):
    async with self.session.client(...) as client:
        yield client

with self.with_client(region) as client:
    # do logic
```

This way you get client open/close work properly out of the box. Calling manually `__aenter__` looks really really dangerous and is in general heavily discouraged in Python.

Reply:
I have added close method in `S3DataSource` class which close the connection to the Amazon S3  and the `client.close()` method  internally calls __aexit__() method . 
for reference:  [https://github.com/aio-libs/aiobotocore/blob/master/aiobotocore/httpsession.py#L170](https://github.com/aio-libs/aiobotocore/blob/master/aiobotocore/httpsession.py#L170%60)

Reply:
Again, manually calling magic Python methods (__aenter__) is considered a bad practice and should be avoided at all costs.